File: erettsegi_emelt/2020_okt/c#/Sorozat.cs
using System;
using System.Globalization;

public class Sorozat {
    public static readonly DateTime HIANYZO_DATUM = DateTime.MinValue;

    public readonly DateTime adasbaKerulesiDatum;
    public readonly string cim;
    public readonly int evadokSzama;
    public readonly int epizodokSzama;
    public readonly int epizodonkentiHossz;
    public readonly bool lattaEMarAKeszito;

    public Sorozat(string[] lines, int index) {
        var datumStr = lines[index];
        var epizodInfoSplit = lines[index + 2].Split('x');

        adasbaKerulesiDatum = datumStr == "NI" ? HIANYZO_DATUM : DateTime.ParseExact(datumStr, "yyyy.MM.dd", CultureInfo.InvariantCulture);
        cim = lines[index + 1];
        evadokSzama = int.Parse(epizodInfoSplit[0]);
        epizodokSzama = int.Parse(epizodInfoSplit[1]);
        epizodonkentiHossz = int.Parse(lines[index + 3]);
        lattaEMarAKeszito = int.Parse(lines[index + 4]) == 1;
    }
}

File: erettsegi_emelt/2020_okt/java/Sorozat.java
import java.time.*;
import java.util.*;

public class Sorozat {
    public static final LocalDate HIANYZO_DATUM = LocalDate.MIN;

    public final LocalDate adasbaKerulesiDatum;
    public final String cim;
    public final int evadokSzama;
    public final int epizodokSzama;
    public final int epizodonkentiHossz;
    public final boolean lattaEMarAKeszito;

    public Sorozat(List<String> lines, int index) {
        var datumStr = lines.get(index);
        var epizodInfoSplit = lines.get(index + 2).split("x");

        adasbaKerulesiDatum = datumStr.equals("NI") ? HIANYZO_DATUM : LocalDate.parse(datumStr.replace('.', '-'));
        cim = lines.get(index + 1);
        evadokSzama = Integer.parseInt(epizodInfoSplit[0]);
        epizodokSzama = Integer.parseInt(epizodInfoSplit[1]);
        epizodonkentiHossz = Integer.parseInt(lines.get(index + 3));
        lattaEMarAKeszito = Integer.parseInt(lines.get(index + 4)) == 1;
    }
}